server/plugin/graphql/lhs_request.go

```diff
@@ -0,0 +1,121 @@
+package graphql
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/go-github/v41/github"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+)
+
+const (
+	queryParamReviewCursor      = "reviewCursor"
+	queryParamAssignmentsCursor = "assignmentsCursor"
+	queryParamOpenPRsCursor     = "openPrsCursor"
+
+	queryParamOpenPRQueryArg   = "prOpenQueryArg"
+	queryParamReviewPRQueryArg = "prReviewQueryArg"
+	queryParamAssigneeQueryArg = "assigneeQueryArg"
+)
+
+func (c *Client) GetLHSData(ctx context.Context) ([]*github.Issue, []*github.Issue, []*github.Issue, error) {
+	params := map[string]interface{}{
+		queryParamOpenPRQueryArg:    githubv4.String(fmt.Sprintf("author:%s is:pr is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
+		queryParamReviewPRQueryArg:  githubv4.String(fmt.Sprintf("review-requested:%s is:pr is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
+		queryParamAssigneeQueryArg:  githubv4.String(fmt.Sprintf("assignee:%s is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
+		queryParamReviewCursor:      (*githubv4.String)(nil),
+		queryParamAssignmentsCursor: (*githubv4.String)(nil),
+		queryParamOpenPRsCursor:     (*githubv4.String)(nil),
+	}
+
+	if c.org != "" {
+		params[queryParamOpenPRQueryArg] = githubv4.String(fmt.Sprintf("org:%s %s", c.org, params[queryParamOpenPRQueryArg]))
+		params[queryParamReviewPRQueryArg] = githubv4.String(fmt.Sprintf("org:%s %s", c.org, params[queryParamReviewPRQueryArg]))
+		params[queryParamAssigneeQueryArg] = githubv4.String(fmt.Sprintf("org:%s %s", c.org, params[queryParamAssigneeQueryArg]))
+	}
+
+	var resultPR, resultAssignee, resultOpenPR []*github.Issue
+	flagPR, flagAssignee, flagOpenPR := false, false, false
+
+	for {
+		if flagPR && flagOpenPR && flagAssignee {
+			break
+		}
+
+		if err := c.executeQuery(ctx, &mainQuery, params); err != nil {
+			return nil, nil, nil, errors.Wrap(err, "Not able to excute the query")
+		}
+
+		if !flagPR {
+			for _, resp := range mainQuery.PullRequest.Nodes {
+				resp := resp
+				pr := getPR(&resp)
+				resultPR = append(resultPR, pr)
+			}
+
+			if !mainQuery.PullRequest.PageInfo.HasNextPage {
+				flagPR = true
+			}
+
+			params[queryParamReviewCursor] = githubv4.NewString(mainQuery.PullRequest.PageInfo.EndCursor)
+		}
+
+		if !flagAssignee {
+			for _, resp := range mainQuery.Assignee.Nodes {
+				resp := resp
```

Comment:
Why?

Reply:
@hanzei To fix this issue 
https://stackoverflow.com/q
![Screenshot from 2023-07-05 20-55-02](https://github.com/mattermost/mattermost-plugin-github/assets/90389917/4828de69-d65d-4d41-b8d4-164ab9338397)
uestions/62446118/implicit-memory-aliasing-in-for-loop

Reply:
The `resp := resp` looks a little off to me. Alternatively we could do this:

```go
for i := range mainQuery.Assignee.Nodes {
    resp := mainQuery.Assignee.Nodes[i]
```

Reply:
sure @mickmister